docs/guide/writing-migrations.md

```diff
@@ -0,0 +1,302 @@
+## Adding migrations
+
+In ApostropheCMS, migrations are added using the [`add(name, fn)` method](/reference/modules/migration.md#add-name-fn) of the `@apostrophecms/migration` module. One common place to add these are within the `init(self)` initialization function of your module. Each migration requires a unique name and is only run **once**. ApostropheCMS tracks which migrations have already been executed, ensuring they won’t run again across restarts or deployments.
+
+While the migration function can be added as an anonymous function as the second argument to the `add()`method, they can also be defined in the `methods(self)` customization function of the module. This can provide for a cleaner `init(self)` function, but is a matter of preference.
+
+Example adding the migration to `init(self)`:
+
+<AposCodeBlock>
+
+```js
+module.exports = {
+  extend: '@apostrophecms/piece-type',
+  async init(self) {
+    // Adding a migration named 'add-featured-to-articles'
+    self.migration.add('add-featured-to-articles', async () => {
+      return self.apos.migration.eachDoc({
+        type: 'article'
+      }, async (doc) => {
+        if (doc.featured === undefined) {
+          await self.apos.doc.db.updateOne({
+            _id: doc._id
+          }, {
+            $set: { featured: false }
+          });
+        }
+      });
+    });
+  }
+};
+```
+<template v-slot:caption>
+  /modules/product/index.js
+</template>
+
+</AposCodeBlock>
+
+Example using `methods(self)`:
+
+<AposCodeBlock>
+
+```js
+module.exports = {
+  extend: '@apostrophecms/piece-type',
+  async init(self) {
+    self.apos.migration.add('add-product-description', self.addProductDescription);
+  }
+  methods(self) {
+    return {
+      async addProductDescription() {
+        await self.apos.migration.eachDoc({
+          type: 'product'
+        }, async (doc) => {
+          if (doc.description === undefined) {
+            await self.apos.doc.db.updateOne({
+              _id: doc._id
+            }, {
+              $set: { description: 'No description available' }
+            });
+          }
+        });
+      };
+    }
+  }
+};
+```
+<template v-slot:caption>
+  /modules/product/index.js
+</template>
+
+</AposCodeBlock>
+
+For both of these examples we are looping through all documents to find the `product` piece types. Then we are using the document `_id` and the Apostrophe database helper method `updateOne` to run the MongoDB operation `$set` that will either create or update the value of the `description` field for that piece. We will go through additional examples in detail below.
+
+### Running Migrations in Production
+
+In production environments, migrations are **not** automatically executed on every startup. This prevents accidental re-running of migrations in stable environments. Instead, migrations must be triggered manually by running a task.
+
+To apply pending migrations in a production environment, run:
+
+```bash
+node app @apostrophecms/migration:run
```

Comment:
1. The task name is `@apostrophecms/migration:migrate`

2. Although migrations currently do run automatically in both development and production, it is best practice to run the `@apostrophecms/migration:migrate` task in production before launching the newest version of the application to serve requests. At a future time, an option to disable Apostrophe's check for needed migrations on ordinary invocations in production may be offered as an optimization.

(Yeaaaaa I just found out this is true two weeks ago sorry LOL)

Reply:
Number 1 - changed. I guess I was rushing.
Number 2 - changed.